### PR TITLE
[WIP] Implementation of QSS functionalities

### PR DESCRIPTION
**Pull request by georgitodorov**

### CODE REVIEW

File: src/common/channelUtils.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ * Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.app");
+
+// XXX find a better way to collect system events
```
Comment (danailbd) [inline src/common/channelUtils.js]: Ask Antranig?

File: src/main/dialog.js [hunk 2]
```diff
@@ -345,3 +372,75 @@ fluid.defaults("gpii.app.i18n.channel", {
...
+            ]
+        },
+        hide: {
+            funcName: "gpii.app.dialog.delayedShow.hide",
+            args: ["{that}"]
+        }
+    }
+});
+
+gpii.app.dialog.delayedShow.show = function (that, delay, showArgs) {
```
Comment (danailbd) [inline src/main/dialog.js]: Here, instead we could override the toggle function for dialog?

File: src/main/gpiiConnector.js [hunk 3]
```diff
@@ -136,6 +136,8 @@ gpii.app.gpiiConnector.parseMessage = function (gpiiConnector, message) {
         var settingPath = path[path.length - 2],
             settingValue = payload.value;
 
+        console.log("SETTING UPDATE: ", settingPath, settingValue);
```
Comment (danailbd) [inline src/main/gpiiConnector.js]: Should we keep this message using the fluid.log() function?

File: src/main/psp.js [hunk 4]
```diff
@@ -129,18 +159,26 @@ fluid.defaults("gpii.app.psp", {
...
+            resizable: false,
+            skipTaskbar: true,
+            backgroundColor: "transparent"
+        }
     },
 
+    // In case we want to have some heightOffset from the screen edge.
+    // This is useful when we have QSS which should be below the PSP.
+    heightOffset: null,
+    offScreenHide: true,
```
Comment (danailbd) [inline src/main/psp.js]: comment

File: src/main/quickSetStrip/qss.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+                    "{gpii.app.qss}.channelListener.events.onQssButtonFocused": {
+                        func: "{that}.showIfPossible",
+                        args: [
+                            "{arguments}.0", // setting
+                            "{arguments}.1"  // metrics
+                        ]
+                    },
+
+                    // TODO we could also have a modelListener and always hide. See widget
+                    "{gpii.app.qss}.events.onDialogHidden": {
```
Comment (danailbd) [inline src/main/quickSetStrip/qss.js]: Idea: way to have a list of events to trigger the same function.

```
compoundListener: {
    hide: {
          funcName: "{that}.hide",
          events: [ ev1, ev2, ...]
    }
}

File: src/main/quickSetStrip/qssDialog.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+gpii.app.qss.show = function (that, params) {
+    if (!that.options.disabled) {
+        // Show the QSS or focus it if it is already shown.
+        if (that.model.isShown) {
+            that.focus();
+        } else {
+            that.applier.change("isShown", true);
+        }
+
+        that.events.onQssOpen.fire(params);
```
Comment (danailbd) [inline src/main/quickSetStrip/qssDialog.js]: we could remove this and simply use the `dialog.events.onDialogShown` event triggered in dialog - `toggle` function (connected with isShown property)

File: src/renderer/qss/css/variables.css [hunk 7]
```diff
@@ -0,0 +1,9 @@
+:root {
+    --main-theme-button-background-color: #83cbef;
```
Comment (danailbd) [inline src/renderer/qss/css/variables.css:2]: make use of this file?
* colors
* sizes

File: src/renderer/qssTooltipPopup/js/qssTooltipPopup.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    /**
+     * Represents the controller for the QSS tooltip that is used to provide
+     * information about the current setting.
+     */
+    fluid.defaults("gpii.qss.qssTooltipPopup", {
+        gradeNames: ["fluid.viewComponent", "gpii.psp.selectorsTextRenderer"],
+
+        model: {
+            messages: {
+                header: "Some header",
```
Comment (danailbd) [inline src/renderer/qssTooltipPopup/js/qssTooltipPopup.js]: clean up

File: src/renderer/qssWidget/js/qssMenuWidget.js [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+            setting: {
+                func: "{channelNotifier}.events.onQssWidgetSettingAltered.fire",
+                args: ["{change}.value"],
+                includeSource: "settingAlter"
+            }
+        },
+        selectors: {
+            menuControls: ".flc-qssMenuWidget-controls"
+        },
+        activationParams: {},
```
Comment (danailbd) [inline src/renderer/qssWidget/js/qssMenuWidget.js]: comments here and for `disabled`

File: src/renderer/qssWidget/js/qssStepperWidget.js [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+    /**
+     * Represents the QSS stepper widget.
+     */
+    fluid.defaults("gpii.qssWidget.stepper", {
+        gradeNames: ["fluid.viewComponent", "gpii.psp.selectorsTextRenderer"],
+
+        model: {
+            messages: {
+                incrementButton: "Larger",
```
Comment (danailbd) [inline src/renderer/qssWidget/js/qssStepperWidget.js]: clear

**Review by danailbd — COMMENTED** [review]
Some initial generic comments

**Comment by georgitodorov** [comment]
Closing as the QSS functionalities have already been merged into master.

**Pull request closed without merging**